go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -90,32 +92,29 @@ def update_files # rubocop:disable Metrics/AbcSize
             run_go_vendor
             run_go_mod_tidy
 
-            # At this point, the go.mod returned from run_go_get contains the
-            # correct set of modules, but running `go get` can change the file
-            # in undesirable ways (such as injecting the current Go version),
-            # so we need to update the original go.mod with the updated set of
-            # requirements rather than using the regenerated file directly
-            original_reqs = original_manifest["Require"] || []
-            updated_reqs = parse_manifest["Require"] || []
-
-            original_paths = original_reqs.map { |r| r["Path"] }
-            updated_paths = updated_reqs.map { |r| r["Path"] }
-            req_paths_to_remove = original_paths - updated_paths
-
-            # Put back the original content before we replace just the updated
-            # dependencies.
-            write_go_mod(original_go_mod)
-
-            remove_requirements(req_paths_to_remove)
-            deps = updated_reqs.map { |r| requirement_to_dependency_obj(r) }
-            update_go_mod(deps)
-
             # put the old replace directives back again
             substitute_all(substitutions.invert)
```

Comment:
The `replace` directives need to be restored _before_ calling `go mod tidy`, or the module graph will be calculated for the wrong versions of modules, which may cause different indirect dependencies.

Reply:
💯 , running `go mod {tidy,vendor}` while Dependabot has stubbed out external modules isn't going to work - I've added a failing test case.

I think instead of inverting the order, Dependabot should only call `go mod tidy` if no modules have been stubbed:
- `go mod tidy` will see different indirect dependencies, as noted
- `go mod vendor` will copy the stubbed module into `vendor/`, which will certainly break downstream CI